frontend/src/slices/journal/journal.slice.ts

```diff
@@ -67,14 +67,12 @@ const { reducer, actions, name } = createSlice({
       state.updateJournalEntryDataStatus = DataStatus.PENDING;
     });
     builder.addCase(updateJournalEntry.fulfilled, (state, action) => {
-      state.allJournalEntries = state.allJournalEntries.map((journalEntry) => {
-        if (journalEntry.id === action.payload.id) {
-          return action.payload;
-        }
-
-        return journalEntry;
-      });
-
+      state.allJournalEntries = state.allJournalEntries.filter(
+        (journalEntry) => {
+          return journalEntry.id !== action.payload.id;
+        },
+      );
+      state.allJournalEntries.unshift(action.payload);
```

Comment:
Please add a middle constant for filtering and adding action.payload on the first position and then pass the value to state

Reply:
Thanks for the advice!